hdfs-scheduler/src/test/java/org/apache/mesos/hdfs/TestHdfsState.java

```diff
@@ -0,0 +1,325 @@
+import org.apache.mesos.Protos.ExecutorInfo;
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.Label;
+import org.apache.mesos.Protos.Labels;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.OfferID;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.SlaveID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskState;
+import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.Protos.Value;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class TestHdfsState {
+  private final Injector injector = Guice.createInjector(new TestSchedulerModule());
+  private final Log log = LogFactory.getLog(TestHdfsState.class);
+  private SecureRandom random = new SecureRandom();
+  private static final String testIdName = "framework";
+  private static final String TEST_HOST = "host";
+  private static final String TEST_TYPE = "type";
+  private static final String TEST_NAME = "name";
+
+  @Test
+  public void testSetFrameworkId()
+    throws ClassNotFoundException, ExecutionException, InterruptedException, IOException {
+    HdfsState state = injector.getInstance(HdfsState.class);
+    FrameworkID expectedId = createFrameworkId();
+    state.setFrameworkId(expectedId);
+    FrameworkID actualId = state.getFrameworkId();
+    assertEquals(expectedId, actualId);
+  }
+
+  @Test
+  public void testRecordTask()
+    throws ClassNotFoundException, IOException, InterruptedException, ExecutionException {
+    HdfsState state = injector.getInstance(HdfsState.class);
+    List<Task> tasks = state.getTasks();
+    assertEquals(0, tasks.size());
+
+    Task inTask = createTask();
+    state.recordTask(inTask);
+    tasks = state.getTasks();
+    assertEquals(1, tasks.size());
+
+    Task outTask = tasks.get(0);
+    assertEquals(inTask.getStatus(), outTask.getStatus());
+    assertEquals(inTask.getOffer(), outTask.getOffer());
+    assertEquals(inTask.getType(), outTask.getType());
+    assertEquals(inTask.getName(), outTask.getName());
+  }
+
+
+  @Test
+  public void testUpdateNoneToLabelsTask()
+    throws ClassNotFoundException, IOException, InterruptedException, ExecutionException {
+    HdfsState state = injector.getInstance(HdfsState.class);
+    List<Task> tasks = state.getTasks();
+    assertEquals(0, tasks.size());
+
+    Task inTask = createTask();
+    state.recordTask(inTask);
+
+    TaskStatus status = createTaskStatusWithLabel(inTask.getId(), TaskState.TASK_RUNNING, "value");
+    state.update(null, status);
+    
+    tasks = state.getTasks();
+    Task outTask = tasks.get(0);
+
+    assertEquals(status, outTask.getStatus());
+  }
+
+  @Test
+  public void testUpdateLabelToLabelTask()
+    throws ClassNotFoundException, IOException, InterruptedException, ExecutionException {
+    HdfsState state = injector.getInstance(HdfsState.class);
+    List<Task> tasks = state.getTasks();
+    assertEquals(0, tasks.size());
```

Comment:
how many times to we do this asserts?  I didn't see the value the first time.. it bloats the code with no zero IMO